### PR TITLE
MAGECLOUD-3231: Rewrite AcceptanceTest to use Cloud Docker

### PR DESCRIPTION
**Pull request by shiftedreality**
### Description

* Update Docker integration framework
* Move Upgrade tests to new framework

### Fixed Issues (if relevant)

1. https://magento2.atlassian.net/browse/MAGECLOUD-3231

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Command/Docker/BuildIntegration.php [hunk 1]
```diff
@@ -78,17 +82,27 @@ protected function configure()
         $this->setName(self::NAME)
             ->setDescription('Build test docker configuration')
             ->addArgument(
-                self::ARGUMENT_PHP,
+                self::ARGUMENT_MODE,
```
Comment (NadiyaS) [inline src/Command/Docker/BuildIntegration.php]: Describe what mode is this. And possible values for it.

File: src/Docker/BuilderFactory.php [hunk 2]
```diff
@@ -13,14 +13,16 @@
 class BuilderFactory
 {
     const BUILDER_DEV = 'dev';
```
Comment (NadiyaS) [inline src/Docker/BuilderFactory.php:15]: Also, not clear enough what every mode does and where to use. You may fully describe it in appropriate class and use @see on necessary places as this

File: src/Docker/DockerIntegrationBuilder.php [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+        $services['services']['build'] = $this->getCliService(
+            $phpVersion,
+            false,
+            ['db'],
+            'build.magento2.docker'
+        );
+        $services['services']['deploy'] = $this->getCliService(
+            $phpVersion,
+            false,
```
Comment (NadiyaS) [inline src/Docker/DockerIntegrationBuilder.php]: why is it false?

File: src/Docker/DockerIntegrationBuilder.php [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+                    'appdata',
+                ],
+                'env_file' => [
+                    './docker/global.env',
+                    './docker/composer.env',
+                ],
+            ]
+        );
+
+        if ($cron) {
```
Comment (billygilbert) [inline src/Docker/DockerIntegrationBuilder.php]: Is this used anywhere?

File: src/Docker/DockerIntegrationBuilder.php (hunk 3 above)
Comment (shiftedreality) [reply src/Docker/DockerIntegrationBuilder.php]: Still under development, Docker engine failing for some reason

File: src/Command/Docker/BuildIntegration.php (hunk 1 above)
Comment (shiftedreality) [reply src/Command/Docker/BuildIntegration.php]: Added

**Review by billygilbert — CHANGES_REQUESTED** [review]
Agree with @NadiyaS.  I'm also curious about the $cron parameter in getCliService.  Even if it's not used, seems more flexible to just make it a generic command string and pass that in instead.

File: src/Command/Docker/BuildIntegration.php (hunk 1 above)
Comment (shiftedreality) [reply src/Command/Docker/BuildIntegration.php]: Changed to `version`. Should be more clear now

File: src/Docker/DockerIntegrationBuilder.php (hunk 4 above)
Comment (shiftedreality) [reply src/Docker/DockerIntegrationBuilder.php]: Added cron container

File: src/Docker/IntegrationV2Builder.php [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    public function __construct(FileList $fileList, ServiceFactory $serviceFactory)
+    {
+        $this->fileList = $fileList;
+        $this->serviceFactory = $serviceFactory;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
```
Comment (NadiyaS) [inline src/Docker/IntegrationV2Builder.php:44]: you may move configurations for fpm, db, web services to separate private methods and collect them here. It'll make code more easy to read/change and you will avoid @SuppressWarning

Comment (shiftedreality) [reply src/Docker/IntegrationV2Builder.php:44]: This is not customer-facing code

Comment (NadiyaS) [reply src/Docker/IntegrationV2Builder.php:44]: yes, we do code more readable for your convenience.. But I don't insist

**Review by billygilbert — APPROVED** [review]

File: tests/docker-integration/Process/Bash.php [hunk 6]
```diff
@@ -35,11 +33,16 @@ public function createCompose(string $command, string $container, array $variabl
...
+    /**
+     * @return array
+     */
+    private function getDefaultVariables(): array
     {
+        $config = new Config();
+
         return [
             'MAGENTO_CLOUD_RELATIONSHIPS' => [
                 'database' => [
```
Comment (NadiyaS) [inline tests/docker-integration/Process/Bash.php:48]: I see these credentials in 3 places. We may think how to move them in one common place to manage. (not required in this task)

**Review by NadiyaS — APPROVED** [review]

File: src/Docker/DockerIntegrationBuilder.php (hunk 3 above)
Comment (BaDos) [reply src/Docker/DockerIntegrationBuilder.php]: @shiftedreality Maybe you need to create a task to remember to change this argument.

**Pull request merged**